server/stream.go

```diff
@@ -605,15 +605,32 @@ func (a *Account) addStreamWithAssignment(config *StreamConfig, fsConfig *FileSt
 	return mset, nil
 }
 
-// Sets the index name. Usually just the stream name and the source's index (so you can source from the same stream more than once)
-// but when the stream is external we will
-// use additional information in case the stream names are the same.
-func (ssi *StreamSource) setIndexName(i int) {
+// Composes  the index name. Contains the stream name, subject filter, and transform destination
+// when the stream is external we will use additional information in case the (external)  stream names are the same.
+func (ssi *StreamSource) composeIName() string {
+	var iName = ssi.Name
+
 	if ssi.External != nil {
-		ssi.iname = strconv.Itoa(i) + ":" + ssi.Name + ":" + getHash(ssi.External.ApiPrefix)
-	} else {
-		ssi.iname = strconv.Itoa(i) + ":" + ssi.Name
+		iName = iName + ":" + getHash(ssi.External.ApiPrefix)
+	}
+
+	filter := ssi.FilterSubject
+	// normalize filter and destination in case they are empty
+	if filter == "" {
```

Comment:
_EMPTY_ vs "" here.

Reply:
Done, sorry I forgot to check for _EMPTY_ and fwcs, it's not a reflex for me yet :)